### ISSUE TITLE
thread.stop() won't kill a thread deadlocked on a intrinsic lock

### ISSUE THREAD

**Issue by quidryan** [jglick/jkillthread#1]
Inherent to Java you can't stop a thread that is blocked on a intrinsic lock, which the exact kind of lock we see our Jenkins lock up on all the time (die die die synchronized method!). There's nothing you could do about it (I think), but you could examine a thread after killing it to see if really did die or not, and then just log it.

https://github.com/jglick/jkillthread/blob/master/src/main/java/com/github/jglick/jkillthread/Agent.java#L51

**Comment by jglick** [comment]
Agreed. In conjunction with #3 we could do something along the lines of

* interrupt
* if it dies with 5s, exit
* stop
* if it dies with 5s, exit
* log a warning
